src/SharedPipelineComponents/DataLockEvents/DeployDataLock/sql/dml/02 datalock.populate.reference.history.sql

```diff
@@ -1,53 +1,263 @@
+/* ----- finish post-op monitoring*/
+
+/* +++++ start pre-op monitoring */
+SELECT @Now = GETDATE()
+SET @operation = 'Pull of all Provider Datalocks from remote server'
+SET @timestamp = CONVERT(VARCHAR(255), @now, 121)
+RAISERROR (		@start,		10,		0,		@operation,		@timestamp		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(),	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish pre-op monitoring*/
+
+INSERT INTO [Reference].[DataLockEvents] (
+	[Id],
+	[DataLockEventId],
+	[ProcessDateTime],
+	[Status],
+	[IlrFileName],
+	[SubmittedDateTime],
+	[AcademicYear],
+	[UKPRN],
+	[ULN],
+	[LearnRefNumber],
+	[AimSeqNumber],
+	[PriceEpisodeIdentifier],
+	[CommitmentId],
+	[EmployerAccountId],
+	[EventSource],
+	[HasErrors],
+	[IlrStartDate],
+	[IlrStandardCode],
+	[IlrProgrammeType],
+	[IlrFrameworkCode],
+	[IlrPathwayCode],
+	[IlrTrainingPrice],
+	[IlrEndpointAssessorPrice],
+	[IlrPriceEffectiveFromDate],
+	[IlrPriceEffectiveToDate]
+	)
+SELECT dle.[Id],
+	dle.[DataLockEventId],
+	dle.[ProcessDateTime],
+	dle.[Status],
+	dle.[IlrFileName],
+	dle.[SubmittedDateTime],
+	dle.[AcademicYear],
+	dle.[UKPRN],
+	dle.[ULN],
+	dle.[LearnRefNumber],
+	dle.[AimSeqNumber],
+	dle.[PriceEpisodeIdentifier],
+	dle.[CommitmentId],
+	dle.[EmployerAccountId],
+	dle.[EventSource],
+	dle.[HasErrors],
+	dle.[IlrStartDate],
+	dle.[IlrStandardCode],
+	dle.[IlrProgrammeType],
+	dle.[IlrFrameworkCode],
+	dle.[IlrPathwayCode],
+	dle.[IlrTrainingPrice],
+	dle.[IlrEndpointAssessorPrice],
+	dle.[IlrPriceEffectiveFromDate],
+	dle.[IlrPriceEffectiveToDate]
+FROM ${DAS_ProviderEvents.FQ}.[DataLock].[DataLockEvents] dle (NOLOCK)
+INNER JOIN @ProvidersToProcess ptp	ON ptp.[UKPRN] = dle.[UKPRN]
+
+/* +++++ start post-op monitoring */
+SET @duration = DATEDIFF(MS, @Now, GETDATE())
+RAISERROR (		@finish,		10,		0,		@operation,		@duration		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(),	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish post-op monitoring*/
+
+/* +++++ start pre-op monitoring */
+SELECT @Now = GETDATE()
+SET @operation = 'Re-Apply Target table Keys, Constrtaints and Indexes'
+SET @timestamp = CONVERT(VARCHAR(255), @now, 121)
+RAISERROR (		@start,		10,		0,		@operation,		@timestamp		) WITH NOWAIT
+INSERT INTO [TestStack].[Logs] (	[LogId],	[Logger],	[LogLevel],	[EntryDate],	[Message],	[ErrorDetails]	)
+SELECT NEWID(),	@logsource,	0,	GETDATE(),	@operation + '- Started: [' + @timestamp + '] Duration: [' + cast(@duration AS VARCHAR(255)) + ']',	NULL
+/* ----- finish pre-op monitoring*/
+
+IF EXISTS (
```

Comment:
Shouldn't this be an IF NOT EXISTS?  Actually, as you've dropped it earlier in the script is the check necessary?

Reply:
Yep, good catch. Last minute changes always are a bad idea! I'll fix and push again. The redundant check is to not rely on assumptions and also for a break in testing.